### PR TITLE
fix(executor): reset access list

### PR DESCRIPTION
**Pull request by rkrasiuk**
Reset access list for legacy transactions

### CODE REVIEW

File: crates/revm/revm-primitives/src/env.rs [hunk 1]
```diff
@@ -81,6 +81,9 @@ pub fn fill_tx_env(tx_env: &mut TxEnv, transaction: &TransactionSigned, sender:
             tx_env.data = input.0.clone();
             tx_env.chain_id = *chain_id;
             tx_env.nonce = Some(*nonce);
+            if !tx_env.access_list.is_empty() {
+                tx_env.access_list.clear();
+            }
```
Comment (mattsse) [inline crates/revm/revm-primitives/src/env.rs:86]: the empty check is not needed

**Review by mattsse — APPROVED** [review]
oh nice catch.

smol nit

File: crates/revm/revm-primitives/src/env.rs (hunk 1 above)
Comment (rkrasiuk) [reply crates/revm/revm-primitives/src/env.rs:86]: agreed

**Review by rakita — APPROVED** [review]
lgtm

**Pull request merged**